src/lavinmq/observable.cr

```diff
@@ -1,19 +1,24 @@
-module Observable
-  @observers = Set(Observer).new
-
-  def register_observer(observer : Observer)
-    @observers.add(observer)
+module LavinMQ
+  module Observer(EventT)
+    abstract def on(event : EventT, data : Object?)
   end
 
-  def unregister_observer(observer : Observer)
-    @observers.delete(observer)
-  end
+  module Observable(EventT)
+    macro included
+      {% ivar = ("@__" + EventT.name.stringify.downcase.gsub(/[^a-z_]+/, "_") + "_observers").id %}
```

Comment:
maybe give it a more explainable name than the generic `ivar`? 

```suggestion
      {% event_observers = ("@__" + EventT.name.stringify.downcase.gsub(/[^a-z_]+/, "_") + "_observers").id %}
```

Reply:
Since we're in macro space I want to keep `ivar` in the name ("ivar" seems to be common short name for "instance variable").